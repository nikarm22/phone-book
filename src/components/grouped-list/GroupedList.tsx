import React from 'react';
import List from '@material-ui/core/List';
import ListSubheader from '@material-ui/core/ListSubheader';

import { makeStyles } from '@material-ui/core/styles';

export interface IGroupedListProps<T> {
  items: { [group: string]: T[] };
  renderer: (item: T) => React.ReactElement;
}

const useStyles = makeStyles((theme) => ({
  root: {
    width: '100%',
    height: '100%',
    backgroundColor: theme.palette.background.paper,
    position: 'relative',
    overflow: 'auto',
  },
  listSection: {
    backgroundColor: 'inherit',
  },
  ul: {
    backgroundColor: 'inherit',
    padding: 0,
  },
}));

export default function GroupedList<T>(props: IGroupedListProps<T>) {
  const { items, renderer } = props;
  const classes = useStyles();

  const groups = Object.keys(items);

  return (
    <List className={classes.root} subheader={<li />}>
      {groups.map((group) => (
        <li className={classes.listSection} key={group}>
          <ul className={classes.ul}>
            <ListSubheader>{group}</ListSubheader>
            {items[group].map((item) => (renderer(item)))}
          </ul>
        </li>
      ))}
    </List>
  );
}
