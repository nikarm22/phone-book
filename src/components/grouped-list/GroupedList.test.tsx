import React from 'react';
import { render } from '@testing-library/react';
import GroupedList from './GroupedList';

const mockItems = {
  groupItem1: [1,2,3],
  groupItem2: [4,5,6]
};

const mockRenderer = (item: number) => <div key={item}>{item}</div>;

describe('components/ContactItem', () => {
  it('should match the snapshot', () => {
    const { asFragment } = render(<GroupedList items={mockItems} renderer={mockRenderer} />);
    expect(asFragment()).toMatchSnapshot();
  })
});
