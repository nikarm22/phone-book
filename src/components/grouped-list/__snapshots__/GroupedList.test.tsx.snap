// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ContactItem should match the snapshot 1`] = `
<DocumentFragment>
  <ul
    class="MuiList-root makeStyles-root-1 MuiList-padding MuiList-subheader"
  >
    <li />
    <li
      class="makeStyles-listSection-2"
    >
      <ul
        class="makeStyles-ul-3"
      >
        <li
          class="MuiListSubheader-root MuiListSubheader-sticky MuiListSubheader-gutters"
        >
          groupItem1
        </li>
        <div>
          1
        </div>
        <div>
          2
        </div>
        <div>
          3
        </div>
      </ul>
    </li>
    <li
      class="makeStyles-listSection-2"
    >
      <ul
        class="makeStyles-ul-3"
      >
        <li
          class="MuiListSubheader-root MuiListSubheader-sticky MuiListSubheader-gutters"
        >
          groupItem2
        </li>
        <div>
          4
        </div>
        <div>
          5
        </div>
        <div>
          6
        </div>
      </ul>
    </li>
  </ul>
</DocumentFragment>
`;
